datacapturing/src/main/java/de/cyface/datacapturing/persistence/MeasurementPersistence.java

```diff
@@ -109,61 +109,66 @@ public void storeData(final @NonNull CapturedData data, final long measurementId
-                                ret.put(RotationPointTable.COLUMN_IS_SYNCED, 0);
-                                ret.put(RotationPointTable.COLUMN_MEASUREMENT_FK, measurementIdentifier);
-                                ret.put(RotationPointTable.COLUMN_TIME, dataPoint.getTimestamp());
-                                return ret;
-                            }
-                        }));
-                operations.addAll(newDataPointInsertOperation(data.getDirections(),
-                        MeasuringPointsContentProvider.MAGNETIC_VALUE_POINTS_URI, new Mapper() {
-                            @Override
-                            public ContentValues map(Point3D dataPoint) {
-                                ContentValues ret = new ContentValues();
-                                ret.put(MagneticValuePointTable.COLUMN_MX, dataPoint.getX());
-                                ret.put(MagneticValuePointTable.COLUMN_MY, dataPoint.getY());
-                                ret.put(MagneticValuePointTable.COLUMN_MZ, dataPoint.getZ());
-                                ret.put(MagneticValuePointTable.COLUMN_IS_SYNCED, 0);
-                                ret.put(MagneticValuePointTable.COLUMN_MEASUREMENT_FK, measurementIdentifier);
-                                ret.put(MagneticValuePointTable.COLUMN_TIME, dataPoint.getTimestamp());
-                                return ret;
-                            }
-                        }));
-
-                for (int i = 0; i < operations.size(); i += MAX_SIMULTANEOUS_OPERATIONS) {
+                ContentValues[] values = new ContentValues[data.getAccelerations().size()];
+                for (int i = 0; i < data.getAccelerations().size(); i++) {
+                    Point3D dataPoint = data.getAccelerations().get(i);
+                    ContentValues ret = new ContentValues();
+                    ret.put(SamplePointTable.COLUMN_AX, dataPoint.getX());
+                    ret.put(SamplePointTable.COLUMN_AY, dataPoint.getY());
+                    ret.put(SamplePointTable.COLUMN_AZ, dataPoint.getZ());
+                    ret.put(SamplePointTable.COLUMN_IS_SYNCED, 0);
+                    ret.put(SamplePointTable.COLUMN_MEASUREMENT_FK, measurementIdentifier);
+                    ret.put(SamplePointTable.COLUMN_TIME, dataPoint.getTimestamp());
+                    values[i] = ret;
+                }
+                for (int i = 0; i < values.length; i += MAX_SIMULTANEOUS_OPERATIONS) {
                     int startIndex = i;
-                    int endIndex = Math.min(operations.size(), i + MAX_SIMULTANEOUS_OPERATIONS);
-                    client.applyBatch(new ArrayList<>(operations.subList(startIndex, endIndex)));
+                    int endIndex = Math.min(values.length,i+MAX_SIMULTANEOUS_OPERATIONS);
+                    client.bulkInsert(MeasuringPointsContentProvider.SAMPLE_POINTS_URI, Arrays.copyOfRange(values, startIndex, endIndex));
```

Comment:
Maybe add a comment, that 'bulkInsert' is faster then 'applyBatch', so this does not happen in the future.